src/webapp/pages/form-page/disease-outbreak-event/utils/diseaseOutbreakEventFormMapper.ts

```diff
@@ -0,0 +1,806 @@
+                                    diseaseOutbreakEventFieldIds
+                                ),
+                                isVisible: true,
+                                errors: [],
+                                type: "text",
+                                value:
+                                    diseaseOutbreakEvent?.earlyResponseActions.responseNarrative ||
+                                    "",
+                                multiline: true,
+                                required: true,
+                                showIsRequired: false,
+                            },
+                        ],
+                    },
+                ],
+            },
+            {
+                title: "Assign Incident Manager",
+                id: "incidentManager_section",
+                isVisible: true,
+                required: true,
+                fields: [
+                    {
+                        id: getFieldIdFromIdsDictionary(
+                            "incidentManagerName",
+                            diseaseOutbreakEventFieldIds
+                        ),
+                        placeholder: "Select a manager",
+                        isVisible: true,
+                        errors: [],
+                        type: "user",
+                        options: teamMemberOptions,
+                        value: diseaseOutbreakEvent?.incidentManagerName || "",
+                        required: true,
+                        showIsRequired: false,
+                    },
+                ],
+            },
+            {
+                title: "Notes",
+                id: "notes_section",
+                isVisible: true,
+                required: false,
+                fields: [
+                    {
+                        id: getFieldIdFromIdsDictionary("notes", diseaseOutbreakEventFieldIds),
+                        isVisible: true,
+                        errors: [],
+                        type: "text",
+                        value: diseaseOutbreakEvent?.notes || "",
+                        multiline: true,
+                    },
+                ],
+            },
+        ],
+    };
+}
+
+function mapToPresentationOptions(options: Option[]): PresentationOption[] {
+    return options.map(
+        (option): PresentationOption => ({
+            value: option.id,
+            label: option.name,
+        })
+    );
+}
+
+export function mapFormStateToEntityData(
+    formState: FormState,
+    currentUserName: string,
+    diseaseOutbreakEventWithOptions: DiseaseOutbreakEventWithOptions
+): DiseaseOutbreakEventBaseAttrs {
+    const { diseaseOutbreakEvent } = diseaseOutbreakEventWithOptions;
+
+    const allFields: FormFieldState[] = getAllFieldsFromSections(formState.sections);
+
+    const hazardType = getStringFieldValue(diseaseOutbreakEventFieldIds.hazardType, allFields);
+
+    const diseaseOutbreakEventEditableData = {
```

Comment:
since FormFieldState has fieldType, can we not have one function which takes allFields and key as the parameter and return the value based on type? 
```

type ValueTypes = string | string [] | Date | boolean ....
getValue(key: fieldName, allFields: FormFieldState[] ) :  ValueTypes {
const field = getFieldValueByName(fieldName)
switch(field.type) {
...
}
}
```

This seems to be easier to understand for me. But will ask you to check with Arnau/Jorge which is better.

Reply:
I can't get it to work properly :(